core/templates/dev/head/exploration_editor/main_tab/StateResponses.js

```diff
@@ -306,6 +314,58 @@ oppia.controller('StateResponses', [
         return choiceIndices.every(function(choiceIndex) {
           return handledAnswersArray.indexOf(choiceIndex) !== -1;
         });
+      } else if (interactionId === 'ItemSelectionInput' &&
+                  maxSelectionCount === 1) {
+        var numChoices = $scope.getAnswerChoices().length;
+        choiceRules = [];
+        for (var i = 0; i < answerGroups.length; i++) {
+          choiceRules.push(false);
+        }
+
+        var newChoiceRules = function(current) {
+          return choiceRules.map(function(element, index) {
+            if (index !== current || element === true) {
+              return true;
+            } else if (index === current && element !== true) {
+              return false;
+            }
+          });
+        };
+
+        for (var i = 0; i < answerGroups.length; i++) {
+          var ruleSpecs = answerGroups[i].rule_specs;
+          for (var j = 0; j < ruleSpecs.length; j++) {
+            for (var k = 0; k < ruleSpecs[j].inputs.x.length; k++) {
+              // If the rule_type of the first input is equalTo or containsOne,
+              // then set first choiceRules to true
+              if (ruleSpecs[j].rule_type === 'Equals') {
+                choiceRules[i] = true;
+              } else if (ruleSpecs[j].rule_type === 'ContainsAtLeastOneOf') {
+                choiceRules[i] = true;
+              } else if (ruleSpecs[j].rule_type ===
+                'DoesNotContainAtLeastOneOf') {
+                // If the ruleType is omits one of, then set the others to true
+                var current = i;
+                choiceRules = newChoiceRules(current);
```

Comment:
Yep. Just tested. The problem I was having previously was this: to get the number of array elements for the for loop, I either had to use numChoices OR answerGroups.length.

In the case of numChoices:
```for (var i = 0; i < numChoices; i++) {
            choiceRules.push(false);
          }
```
Then, I would get an array of a length that was equal to the number of answer choices. This worked fine except in cases where I had more answer groups then I had choices (so for example, I had 3 answer choices but had written 5 answer groups). In that case, extra answer groups that contained the DoesNotContainAtLeastOneOf rule type and would, therefore, create a false boolean, were ignored because the choiceRules array would only ever have 3 items. Conversely, if I used DoesNotContainAtLeastOneOf in the first three answers but had extra answers that did, in fact cover those answer groups, the choiceRules array would never see those extra answer choices and would continue having a false boolean and showing an error.

My other option was to use answerGroups.length:
```
for (var i = 0; i < answerGroups.length; i++) {
            choiceRules.push(false);
          }
```
But the problem with that option is of course, that if I have 3 answer choices but only one answer group and that answer group was covered, (so choiceRules contained a true) then it would suppress the warning.

So I was using both the handledAnswers array (which checked that the text of each answer choice showed up) and the choiceRules array (which checked the rule type of each answer group to check whether each choice was covered).

The other thing that's possible is that, I can have three answer groups that cover the same answer choice, so I would have a choiceRules array with three true values (which would suppress the warning) but that only covered one answer choice. There is still a warning on the exploration overview map about clarifying the default outcome. But if I engage both the handledAnswers and choiceRules arrays, then the warning is not suppressed.